File: src/main/java/com/example/dto/response/branch/BranchDetailResponse.java
package com.example.dto.response.branch;

import com.example.dto.response.address.AddressDetailResponse;
import com.example.dto.response.customer.CustomerResponse;
import com.example.dto.response.institute.InstituteResponse;

import java.util.Set;

public record BranchDetailResponse(Long id, String branchCode, String locationCode, String bicCode, String name,
                                   String email, String phoneNumber, AddressDetailResponse address,
                                   InstituteResponse institute, Set<CustomerResponse> customers,
                                   String lastModifiedDate, String createdDate, Long version) {
}

File: frontend/src/app/core/models/response/branch/branch-detail-response.ts
import {InstituteResponse} from '../institute/institute.response';
import {AddressResponse} from '../address/address-response';
import {CustomerResponse} from '../customer/customer-response';
import {BranchResponse} from './branch-response';

export interface BranchDetailResponse extends BranchResponse {

  address: AddressResponse;
  institute: InstituteResponse;
  customers: CustomerResponse[];
  createdDate: string;
  version: number;

}
